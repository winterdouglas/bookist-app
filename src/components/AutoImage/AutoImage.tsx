import { useAutoImage } from "@/hooks/useAutoImage";
import { forwardRef } from "react";
import {
  Image,
  ImageProps as RNImageProps,
  ImageURISource,
} from "react-native";

export type AutoImageProps = RNImageProps & {
  maxWidth?: number;
  maxHeight?: number;
};

/**
 * An Image component that automatically sizes a remote or data-uri image.
 * - [Documentation and Examples](https://github.com/infinitered/ignite/blob/master/docs/Components-AutoImage.md)
 */
export const AutoImage = forwardRef<Image, AutoImageProps>((props, ref) => {
  const { maxWidth, maxHeight, ...ImageProps } = props;
  const source = props.source as ImageURISource;

  const [width, height] = useAutoImage(source?.uri, [maxWidth, maxHeight]);

  return (
    <Image ref={ref} {...ImageProps} style={[{ width, height }, props.style]} />
  );
});
