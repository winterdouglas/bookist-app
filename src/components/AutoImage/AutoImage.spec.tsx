import renderer from "react-test-renderer";
import { AutoImage } from "./AutoImage";

it("renders correctly", () => {
  const tree = renderer
    .create(<AutoImage source={{}} maxWidth={100} maxHeight={200} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
