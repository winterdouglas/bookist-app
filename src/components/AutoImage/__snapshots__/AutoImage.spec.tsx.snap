// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Image
  source={{}}
  style={
    [
      {
        "height": 0,
        "width": 0,
      },
      undefined,
    ]
  }
/>
`;
